src/views/ToDoList/ToDoList.test.tsx

```diff
@@ -0,0 +1,262 @@
+import { vi, Mock } from 'vitest';
+import { Task } from '../../shared/types/Task';
+import { toDosApi } from '../../shared/api/toDosApi';
+import { ToDoList } from './index';
+import { fireEvent, render } from '@testing-library/react';
+
+vi.mock('../../shared/api/toDosApi', () => {
+  return {
+    toDosApi: {
+      fetchToDos: vi.fn(),
+      deleteTask: vi.fn(),
+      createNewTask: vi.fn(),
+    },
+  };
+});
+
+describe('The ToDoList', () => {
+  let newTaskTitle: string;
+  let newTaskMock: Task;
+  let task1: Task;
+  let task2: Task;
+  let task3: Task;
+  let task4: Task;
+  let returnedTasksList: Task[];
+
+  beforeEach(() => {
+    task1 = {
+      userId: 1,
+      id: 1,
+      title: 'delectus aut autem',
+      completed: true,
+    };
+    task2 = {
+      userId: 1,
+      id: 2,
+      title: 'quis ut nam facilis et officia qui',
+      completed: false,
+    };
+    task3 = {
+      userId: 1,
+      id: 3,
+      title: 'fugiat veniam minus',
+      completed: false,
+    };
+    task4 = {
+      userId: 1,
+      id: 4,
+      title: 'qui ullam ratione quibusdam voluptatem quia omnis',
+      completed: true,
+    };
+
+    returnedTasksList = [task1, task2, task3, task4];
+    newTaskTitle = 'mocked task';
+    newTaskMock = {
+      title: newTaskTitle,
+      id: 1,
+      userId: 1,
+      completed: false,
+    };
+
+    (toDosApi.fetchToDos as Mock).mockReturnValue(new Promise(() => null));
+  });
+
+  describe('when the component is mounted', () => {
+    describe('and the Api call is successful', () => {
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+      });
+      it('should display tasks and not display error', async () => {
```

Comment:
- displaying the created tasks
- not displaying an error
are two separate test cases and should have two separate its